### PR TITLE
feat: show progressbar for tt pull & push

### LINKED ISSUE #3: Progress bar for pushing and pulling documents
```
/config/locales/foo.yml
[###################################] 100%
```

### PR DESCRIPTION
**Pull request by lukad**
Closes #3 

- [x] Implement `Documents#download_url` (Toptranslation/toptranslation_ruby#20, Toptranslation/toptranslation_ruby#21)
- [x] Release Toptranslation/toptranslation_ruby#21
- [x] Progressbar for `tt pull`
- [x] Return Content-Length from download endpoint
- [x] Fix clashing files in file compose service
- [x] Make sure the documents translations endpoint returns a SHA1 hash
- [x] Refactor progressbar code
- [x] Progressbar for `tt push`

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,3 +3,6 @@
 source 'https://rubygems.org'
 
 gemspec
+
+# TODO: remove
+gem 'toptranslation_api', path: '../toptranslation_ruby'
```
Comment (lukad) [inline Gemfile]: Remove this.

File: Gemfile [hunk 2]
```diff
@@ -3,3 +3,5 @@
 source 'https://rubygems.org'
 
 gemspec
+
+gem 'toptranslation_api', path: '../toptranslation_ruby'
```
Comment (lukad) [inline Gemfile]: This can be removed when Toptranslation/toptranslation_ruby#23 is merged.

File: lib/toptranslation_cli/pull.rb [hunk 3]
```diff
@@ -48,6 +60,27 @@ def project_locales
...
+              document: document,
+              sha1: translation.sha1,
+              locale: translation.locale
+            }
+          end
+        end
+      end
+
+      def download(file, bar)
+        file[:document].download(file[:locale].code, path: file[:path], file_type: 'yaml') do |n, total|
```
Comment (sr189) [inline lib/toptranslation_cli/pull.rb]: Why does it say `yaml` there? Did we fix the filetype somewhen for debugging?

**Review by sr189 — COMMENTED** [review]
It's easier for me to see what can be refactored after it's merged. Hard for me to see it in a diff-view. Except for the fixed `yaml` this looks ok to me.

File: lib/toptranslation_cli/pull.rb (hunk 3 above)
Comment (lukad) [reply lib/toptranslation_cli/pull.rb]: Good question. It was yaml before, so I kept it: https://github.com/Toptranslation/tt_cli/pull/11/files#diff-ecc4120ae09237ec53d471dc646014feL12

**Review by Farson — APPROVED** [review]

**Pull request merged**